src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num !== undefined) {
+      sum += num;
+
+      return adder;
+    } else {
+      const currentSum = sum;
+
+      sum = 0;
+
+      return currentSum;
+    }
```

Comment:
The implementation of the `adder` function is correct, but it's worth noting that the function will not work as expected if called with a falsy value that is not `undefined`, such as `0`. Since the task description does not require handling this case, the current implementation meets the requirements.